src/components/Checkbox/Checkbox.tsx

```diff
@@ -55,8 +55,8 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
     ) {
       throw new Error('You must provide a visible label or aria-label');
     }
-    const generatedId = useUID();
-    const checkboxId = id || generatedId;
+    const generatedId = useId(id);
+    const checkboxId = String(generatedId);
```

Comment:
The `htmlFor` prop will complain because `generatedId` can be null, so we cast it so a string instead